### DISCUSSION TITLE
send pictures/documentation and more

### DISCUSSION THREAD

**Discussion by tokensave** [def-studio/telegraph#524] [Q&A]
Hello, I can’t figure out how to send a picture to the chat from a bot when calling a certain command
Used different methods:
    public function test(){
1)$this->chat->message('blabla')->photo(Storage::path('first.jpg'))->send();
or
2)Telegraph::photo(Storage::path('first.jpg'))->send();

in the first case I didn’t send anything at all
in the second there is an error: No TelegraphChat defined for this request

Maybe I can install something in the config, there’s nothing specifically described in the documentation for this, or maybe I don’t see what.
help me please

**Comment by fabio-ivona** [comment]
Hi, the `1)` is th correct one

you could use the `2)` as well, but you should add the chat where you want to send the photo 

```php
Telegraph::chat(<chat>)->photo(Storage::path('first.jpg'))->send();
```

In order to help you, could you paste here the dump of the response?

```php
$this->chat->photo(Storage::path('first.jpg'))->send();
```

**Reply by tokensave** [reply]
local.DEBUG: test command [{"DefStudio\\Telegraph\\Client\\TelegraphResponse":"{\"ok\":false,\"error_code\":400,\"description\":\"Bad Request: invalid file HTTP URL specified: URL host is empty\"}"}]

**Reply by tokensave** [reply]
public function test()
    {
        $this->chat->message('blabla')->photo(Storage::path('first.jpg'))->send();
    }
    
    file in 
![Снимок экрана 2024-02-22 115829](https://github.com/defstudio/telegraph/assets/110473226/d6e8df40-9a10-43fc-8815-0eb3d7c345ec)

**Reply by tokensave** [reply]
test in local with localhost.run 
it give me https url

**Reply by tokensave** [reply]
Perhaps I should give a link to the photo, and not send it from my local storage? although I see that the documentation is sending from local storage

**Reply by fabio-ivona** [reply]
images from local storage is supported as well

could you please paste the result of this? (request dump)

```php
$this->chat->photo(Storage::path('first.jpg'))->dump()
```

**Reply by tokensave** [reply]
"DefStudio\\Telegraph\\Telegraph":[]

**Reply by tokensave** [reply]
```
    public function photo(string $path, string $filename = null): self
    {
        $telegraph = clone $this;

        $telegraph->endpoint = self::ENDPOINT_SEND_PHOTO;

        $telegraph->data['chat_id'] = $telegraph->getChatId();
        
        $this->attachPhoto($telegraph, $path, $filename);

        return $telegraph;
    }
```

for some reason this place is $telegraph also empty

$this->chat->message('blabla')->photo(Storage::path('public/first.jpg'))->dump();

**Reply by tokensave** [reply]
i understand my error
        $a = $this->chat->message('blabla')->photo(Storage::path('public/first.jpg'), 'lala.jpg')->send();
i dont give name my file
